sips/sip-merge-staked-sui.md

```diff
@@ -0,0 +1,133 @@
+| Author              | ripleys <0xripleys@solend.fi> |
+| Editor              | Amogh Gupta <amogh@sui.io, @amogh-sui>  |
+| Type                | Standard |
+| Category            | Framework |
+| Created             | 2024-05-23 |
+| Comments-URI        | https://sips.sui.io/comments-31 |
+| Status              | Review |
+| Requires            | NA |
+
+## Abstract
+
+Allow StakedSui objects to be transformed into an Fungible object, which are fungible and epoch independent. This improves efficiency for anyone working with a large amount of StakedSui objects, primarily liquid staking derivative contracts.
+
+## Motivation
+
+Currently, StakedSui objects are tied to a specific activation epoch. If two StakedSui objects are from different activation epochs, they cannot be merged together. 
+
+This is annoying for anyone working with a large amount of StakedSui objects, primarily liquid staking derivative (LST) contracts. Currently, any LST contract now has to manage O(n) StakedSui objects _per_ validator, where n is the number of epochs since genesis. See [the vSui implementation](https://github.com/Sui-Volo/volo-liquid-staking-contracts/blob/main/liquid_staking/sources/validator_set.move#L45) for reference. This is inefficient, as any LST unstake operation can now touch a large amount of StakedSui objects.
+
+## Specification
+
+[staking_pool.move](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/packages/sui-system/sources/staking_pool.move)
+
+```move
+
+    /// An FungibleStakedSui object with a value of 1 corresponds to 1 pool token in the Staking pool.
+    /// This can be a Coin! See the Rationale below.
+    public struct FungibleStakedSui has key, store {
+        id: UID,
+        /// ID of the staking pool we are staking with.
+        pool_id: ID,
+        /// The pool token amount.
+        value: u64,
+    }
+
+
+    /// Dynamic field on the StakingPool Struct.
+    public struct FungibleStakedSuiData has key, store {
+        id: UID,
+        /// fungible_stake supply. sum of values across all FungibleStakedSui objects in the pool.
+        fungible_stake_supply: u64,
+        /// principal balance. Rewards are not stored here, they are withdrawn from the StakingPool's reward pool.
+        principal: Balance<SUI>,
+    }
+
+    // === dynamic field keys ===
+    public struct FungibleStakedSuiDataKey has copy, store, drop {}
+
+    // === Public getters ===
+    public fun fungible_stake_value(fungible_stake: &FungibleStakedSui): u64 {
+        fungible_stake.value
+    }
+
+    public fun fungible_stake_pool_id(fungible_stake: &FungibleStakedSui): ID {
+        fungible_stake.pool_id
+    }
+
+    public fun fungible_stake_to_sui_amount(pool: &StakingPool, fungible_stake_amount: u64): u64;
+    public fun sui_to_fungible_stake_amount(pool: &StakingPool, sui_amount: u64): u64;
+
+    public fun join_fungible_stake(self: &mut FungibleStakedSui, other: FungibleStakedSui);
```

Comment:
Great SIP overall!

Can we add method aliases for the `join_fungible_stake` and `split_fungible_stake` functions?

Reply:
yeah, can definitely do that.